src/index.ts

```diff
@@ -449,6 +452,122 @@ export class Datastore {
         await this._contract.removeGroupFromFile(fileId, groupId)
     }
 
+    /**
+     * Add a new label to the Datastore
+     * @param {string} name Label name
+     * @param {string} color Label color
+     */
+    async createLabel(name: string, color: string) {
+        await this._initialize()
+
+        if (name.length > 28)
+            throw 'Label name must not exceed 28 characters.'
+        
+        let hexColor = Color(color).hex()
+        await this._contract.createLabel(name, hexColor)
+    }
+
+    /**
+     * Delete a label from the Datastore
+     * @param {number} labelId Label Id
+     */
+    async deleteLabel(labelId: number) {
+        await this._initialize()
+
+        await this._contract.deleteLabel(labelId)
+    }
+
+    /**
+     * Assign a label to a file
+     * @param {number} fileId File Id
+     * @param {number} labelId Label Id
+     */
+    async assignLabel(fileId: number, labelId: number) {
+        await this._initialize()
+
+        await this._contract.assignLabel(fileId, labelId)
+    }
+
+    /**
+     * Unassign a label from a file
+     * @param {number} fileId File Id
+     * @param {number} labelIdPosition Label Id's position
+     */
+    async unassignLabel(fileId: number, labelId: number) {
+        await this._initialize()
+
+        let fileLabels = await this._contract.getFileLabelList(fileId)
+        let labelIdPosition
+        for (let i = 0; i < fileLabels.length; i++) {
```

Comment:
You can probably replace the loop with 
```suggestion
        const labelIdPosition = fileLabels.findIndex(id => id === labelId)
```